src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/custom.py

```diff
@@ -0,0 +1,241 @@
+            policy_states_resource,
+            default_subscription_id,
+            resource_group_name,
+            query_options)
+    elif management_group_name is not None:
+        states = client.list_query_results_for_management_group(
+            policy_states_resource,
+            management_group_name,
+            query_options)
+    else:
+        states = client.list_query_results_for_subscription(
+            policy_states_resource,
+            default_subscription_id,
+            query_options)
+
+    return states.value
+
+
+def summarize_policy_states(
+        cmd,
+        client,
+        management_group_name=None,
+        subscription_id=None,
+        resource_group_name=None,
+        resource_id=None,
+        policy_set_definition_name=None,
+        policy_definition_name=None,
+        policy_assignment_name=None,
+        from_value=None,
+        to_value=None,
+        top_value=None,
+        filter_clause=None):
+
+    from azure.mgmt.policyinsights.models import QueryOptions
+
+    query_options = QueryOptions(
+        top=top_value,
+        from_property=from_value,
+        to=to_value,
+        filter=filter_clause)
+
+    default_subscription_id = subscription_id
+    if default_subscription_id is None:
+        default_subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if policy_assignment_name is not None:
+        if resource_group_name is not None:
+            summary = client.summarize_for_resource_group_level_policy_assignment(
+                default_subscription_id,
+                resource_group_name,
+                policy_assignment_name,
+                query_options)
+        else:
+            summary = client.summarize_for_subscription_level_policy_assignment(
+                default_subscription_id,
+                policy_assignment_name,
+                query_options)
+    elif policy_definition_name is not None:
+        summary = client.summarize_for_policy_definition(
+            default_subscription_id,
+            policy_definition_name,
+            query_options)
+    elif policy_set_definition_name is not None:
+        summary = client.summarize_for_policy_set_definition(
+            default_subscription_id,
+            policy_set_definition_name,
+            query_options)
+    elif resource_id is not None:
+        summary = client.summarize_for_resource(
+            resource_id,
+            query_options)
+    elif resource_group_name is not None:
+        summary = client.summarize_for_resource_group(
+            default_subscription_id,
+            resource_group_name,
+            query_options)
+    elif management_group_name is not None:
+        summary = client.summarize_for_management_group(
+            management_group_name,
+            query_options)
+    else:
+        summary = client.summarize_for_subscription(
+            default_subscription_id,
+            query_options)
+
+    return summary.value[0]
```

Comment:
What is the difference between summary and list?

Reply:
List returns a list of policy states. Summarize returns pre-cooked aggregations as a summary object. Feel free to reach out to me if you need more info on the RP. I explained the RP to @williexu briefly.